File: experiments-laptop/ChatGPT_haskell_PS_EE/code.hs
import qualified Data.Map as Map
import Data.Maybe (fromJust)
import System.IO (readFile)
import Text.Read (readMaybe)
import Control.Monad (when)

-- Function to find two indices that sum up to the target
twoSum :: [Int] -> Int -> (Int, Int)
twoSum nums target = go nums Map.empty 0
  where
    -- Helper function to iterate through the list
    go :: [Int] -> Map.Map Int Int -> Int -> (Int, Int)
    go [] _ _ = error "No solution found"
    go (x:xs) map index = 
      let complement = target - x
      in case Map.lookup complement map of
           Just complementIndex -> (complementIndex, index)
           Nothing -> go xs (Map.insert x index map) (index + 1)

-- Main function for testing
main :: IO ()
main = do
  -- Read the contents of the file "input.txt"
  content <- readFile "input.txt"
  
  -- Split the content into lines
  let (targetLine:numsLine:_) = lines content
  
  -- Read the target
  let maybeTarget = readMaybe targetLine :: Maybe Int
  
  -- Read the numbers from the second line
  let maybeNums = mapM (readMaybe :: String -> Maybe Int) (words numsLine)
  
  case (maybeTarget, maybeNums) of
    (Just target, Just nums) -> do
      let result = twoSum nums target
      putStrLn $ show result
    _ -> putStrLn "Error: Invalid input format in input.txt."

File: experiments-laptop/Claude_cpp_PS_OG/code.cpp
#include <vector>
#include <unordered_map>
#include <iostream>
#include <fstream>
#include <sstream>


using namespace std;

class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> numMap;
        
        for (int i = 0; i < nums.size(); i++) {
            int complement = target - nums[i];
            
            if (numMap.find(complement) != numMap.end()) {
                return {numMap[complement], i};
            }
            
            numMap[nums[i]] = i;
        }
        
        // If no solution is found
        return {};
    }
};

int main() {
    std::ifstream infile("input.txt");  // Read from input.txt
    std::string line;
    
    // Read the target
    std::getline(infile, line);
    int target = std::stoi(line);
    
    // Read the vector elements
    std::vector<int> nums;
    std::getline(infile, line);
    std::istringstream iss(line);
    int num;
    while (iss >> num) {
        nums.push_back(num);
    }

    // Your Two Sum logic
    Solution solution;
    std::vector<int> result = solution.twoSum(nums, target);
    
    if (!result.empty()) {
        std::cout << result[0] << ", " << result[1] << std::endl;
    } else {
        std::cout << "No solution found." << std::endl;
    }

    return 0;
}